src/instructlab/configuration.py

```diff
@@ -147,6 +157,11 @@ def api_base(self):
         return get_api_base(self.host_port)
 
 
+class _train(BaseModel):
+    train_args: TrainingArgs
+    torch_args: TorchrunArgs
```

Comment:
I don't consider this blocking but note that there is an API encapsulation problem here. The ilab CLI project governs the config file format, yet it's using a library dep to know how to parse the configuration and parameters. Any sort of change to the library API class will silently fail in the CLI code, and alter the CLI contract (issue new values etc).  Although this would be fine if it was part of the same code base. A common strategy to deal with this is to have a mapping class that copies the values across.

Reply:
I actually just had this convo with Nathan. I am going to investigate having our own types that mirror the training ones but need to be manually updated in a future PR.